### PR TITLE
Barebones working version of gocd-mesos framework

### PR DESCRIPTION
**Pull request by tamizhgeek**

### CODE REVIEW

File: src/main/resources/application.conf [hunk 1]
```diff
@@ -1,9 +1,20 @@
...
+  # Mesos master URL. Should be accessible from where the framework is running. The mesos master should
+  # also able to be talk to framework. The framework should bind to a public or visible interface to mesos-master using LIBPROCESS_IP
+  mesos-master: "192.168.33.10:5050"
+
+  # GO server config to poll and connect the agent
+  go-server {
+    host: "localhost"
+    port: "8080"
+    user-name: "azhagu"
+    password: "random-insecure-string"
```
Comment (brewkode) [inline src/main/resources/application.conf:11]: Well, not really random :P

File: src/test/scala/com/indix/mesos/GoCDPollerTest.scala [hunk 2]
```diff
@@ -0,0 +1,63 @@
+package com.indix.mesos
+
+import com.typesafe.config.ConfigFactory
+import org.scalatest._
+import org.mockito.Mockito._
+
+class GoCDPollerTest extends FlatSpec with Matchers with BeforeAndAfterEach {
```
Comment (brewkode) [inline src/test/scala/com/indix/mesos/GoCDPollerTest.scala]: Test classes are better named with Spec suffix.

**Comment by brewkode** [comment]
Azhagu - Refreshing my memory, what was the specific issue we had with the DockerExecutor that made us to move to CommandExecutor..? Or, was it lack of bandwidth to get a proper Docker Image itself?

**Comment by brewkode** [comment]
gocd-docker containers: https://github.com/gocd/gocd-docker

**Comment by tamizhgeek** [comment]
@sriram-r Bandwidth is one reason. Also, the official gocd docker build never worked even when I tried recently. The travix/gocd-agent one worked like a charm :)

**Comment by brewkode** [comment]
@tamizhgeek Thanks for reviving this. :+1:

**Comment by tamizhgeek** [comment]
Thanks @sriram-r and @ashwanthkumar. Merging this.

**Pull request merged**